dart/gui/osg/Viewer.hpp

```diff
@@ -288,6 +293,13 @@ class Viewer : public osgViewer::Viewer, public dart::common::Subject
   /// Get the root ::osg::Group of this Viewer
   const ::osg::ref_ptr<::osg::Group>& getRootGroup() const;
 
+  /// Get the physics root ::osg::Group of this Viewer
+  const ::osg::ref_ptr<::osg::Group>& getPhysicsGroup() const;
+
+  bool isShadowed() const;
+  void enableShadows(bool _enable = true, ::osg::ref_ptr<osgShadow::ShadowTechnique> shadowTechnique = nullptr);
```

Comment:
These default values seem very counter-intuitive to me. Is it possible to have shadows with a `nullptr` for the `osgShadow::ShadowTechnique`? I would assume that providing a `nullptr` will make it impossible for OSG to render shadows, so it wouldn't make sense to have them enabled. Or is the `shadowTechnique` something extra that can be applied on top of the default OSG shadowing?

Reply:
> These default values seem very counter-intuitive to me. Is it possible to have shadows with a nullptr for the osgShadow::ShadowTechnique? I would assume that providing a nullptr will make it impossible for OSG to render shadows, so it wouldn't make sense to have them enabled. Or is the shadowTechnique something extra that can be applied on top of the default OSG shadowing?

You need the `ShadowTechnique` in order to have shadows..

Reply:
Right, which is why this API seems deceptive. It's simultaneously telling the Viewer "I want you to enable shadows" and "I don't want you to have a `ShadowTechnique`.

I understand now that we're interpreting a `nullptr` as "use some default `ShadowTechnique`, but that isn't clear without digging into the source code of this function.